numpy/core/tests/test_item_selection.py

```diff
@@ -79,9 +79,29 @@ def test_empty_partition(self):
         assert_array_equal(a, a_original)
 
     def test_empty_argpartition(self):
-            # In reference to github issue #6530
-            a = np.array([0, 2, 4, 6, 8, 10])
-            a = a.argpartition(np.array([], dtype=np.int16))
+        # In reference to github issue #6530
+        a = np.array([0, 2, 4, 6, 8, 10])
+        a = a.argpartition(np.array([], dtype=np.int16))
+
+        b = np.array([0, 1, 2, 3, 4, 5])
+        assert_array_equal(a, b)
+
+    def test_nonzero_safety(self):
+        # from Github isue #13631
+        class FalseThenTrue:
+            _val = False
+            def __bool__(self):
+                try: return self._val
+                finally: self._val = True
+
+        try:
+            np.array([True, FalseThenTrue()]).nonzero()
+            assert_(False)
```

Comment:
Probably might as well use `assert` nowadays, not sure I prefer `pass` instead of `assert True`.

Reply:
This should be using
```
with pytest.assert_raises(RuntimeError):
    np.array([True, FalseThenTrue()]).nonzero()
```